### PR TITLE
lib/parser: Use project for G_OPT_M_LOCATION

### PR DESCRIPTION
**Pull request by wenzeslaus**
Standard option G_OPT_M_LOCATION which creates option named location now creates option named project. Each module using G_OPT_M_LOCATION without further modifications has its own entry in in the renamed options file for backwards compatibility.

New usage of _g.mapset_:

```
> g.mapset project=nc_basic_spm_grass7 mapset=user1
```

Still works:

```
> g.mapset location=nc_spm_08_grass7 mapset=user1
```

The latter produces a warning:

```
WARNING: Please update the usage of <g.mapset>: option <location> has been
         renamed to <project>
```

`g.mapset --h`:

```
Changes/reports current mapset.

Usage:
 g.mapset [-clp] mapset=name [project=name] [dbase=path] [--overwrite]
   [--help] [--verbose] [--quiet] [--ui]

Flags:
  -c   Create mapset if it doesn't exist
  -l   List available mapsets and exit
  -p   Print current mapset and exit

Parameters:
   mapset   Name of mapset (default: current search path)
  project   Project (location) name
    dbase   GRASS GIS database directory
```

This applies to:

* g.mapset
* r.proj
* i.ortho.elev
* g.download.location
* v.proj

### CODE REVIEW

**Comment by wenzeslaus** [comment]
Yes. That fits with the general plan of adding "project" to the API and then gradually removing usage of "location" internally.

**Review by cmbarton — CHANGES_REQUESTED** [review]
Note the inconsistencies in how the terms Mapset and Location are written in this and other places: mapset, MAPSET, and Mapset. Since we are changing terminology, this is a good opportunity to standardize how these important terms are written. 

As suggested in #2993 and other places, I suggest that we capitalize the first letter and use lower case for the rest of the letters in all descriptions, labels, and documentation--i.e., Mapset and Project--wherever possible. This reduces confusion when people read about these important terms. 

For command-line arguments, I suggest keeping everything in lower case letters of course.

File: lib/gis/renamed_options [hunk 1]
```diff
@@ -718,3 +718,10 @@ v.what.vect|qvector:query_map
 v.what.vect|qlayer:query_layer
 v.what.vect|qcolumn:query_column
 v.what.vect|vector:map
+### Renamed options for location to project rename (8.3->8.4)
+#############################################################
+g.mapset|location:project
+r.proj|location:project
+i.ortho.elev|location:project
+g.download.location|location:project
+v.proj|location:project
```
[the remark this answers is not in the document]
Comment (wenzeslaus) [reply lib/gis/renamed_options:727]: Usage updated in grass.jupyter and elsewhere.

File: raster/r.proj/r.proj.html [hunk 2]
```diff
@@ -206,7 +206,7 @@ <h3>Inline method</h3>
 
 <div class="code"><pre>
 # calculate where output map will be
-r.proj input=elevation location=ll_wgs84 mapset=user1 -p
```
[the remark this answers is not in the document]
Comment (wenzeslaus) [reply raster/r.proj/r.proj.html:209]: The page is long and has a lot of occurrences of location. Perhaps better handled separately from the change in code and usage.

**Review by petrasovaa — APPROVED** [review]

**Pull request merged**